### DISCUSSION TITLE
How to use :show-help when defining a new prefix?

### DISCUSSION THREAD

**Discussion by Munksgaard** [magit/transient#309] [Support] [answered]
I'm trying to define a new prefix for a command that doesn't have a man-page, but I'm having trouble figuring out how `:show-help` is supposed to be used. Ideally, I'd just want it to show the output of `mycommand --help` in a buffer, but I can't find any examples anywhere of how to do that.

This is what I have so far:

```elisp
(transient-define-prefix gt-create ()
  "Create a new branch"
  :show-help (lambda (obj) (debug(obj)))
  ["Arguments"
   ("-a" "Stage all unstaged changes before creating the branch, including to untracked files." "--all")]
  ["Actions"
   ("a" "Create" gt-create)]
  (interactive)
  (transient-setup 'gt-create))
```

Which gives me the following output in my messages:

```
transient-help: (void-function obj)
Unbound suffix: ‘?’ (Use ‘C-g’ to abort, ‘?’ for help) [transient-help]
```

**Comment by tarsius** [comment, marked as answer]
The error message is due to a defect in your code that has nothing to do with Transient:

```elisp
(lambda (obj) (debug (obj))
==error==> (void-function obj)
```

`obj` is in "function position" and thus called as a function, but there is no function named `obj` here, just a variable by that name.

What you probably wanted to do is:

```elisp
:show-help (lambda (obj) (message "Let's test what obj looks like: %S" obj))
```

**Comment by Munksgaard** [comment]
You're right, thank you very much.

**Reply by tarsius** [reply]
(Meta: Note that there are multiple "reply text areas". In this case you should have used the other one.)

Last night I've written some code to make this sort of thing easier. I'm out the door now, but will finish it later.

**Reply by Munksgaard** [reply]
Amazing! I'll look forward to it.

For context, I am trying to write some (very light) magit-like functionality for the [Graphite CLI](https://graphite.dev/docs/graphite-cli), so I'm basically just trying to show the result of e.g. `gt create --help`.

**Comment by tarsius** [comment]
If added a new macro for this in 5fcd713873001724c8b21d3689c77cb7a6dc9cd7, along with some documentation. A basic example:

```elisp
(transient-define-prefix demo ()
  :show-help (lambda (_)
               (transient-with-help-window
                 (process-file "ls" nil t nil "--help")))
  [(transient-echo-arguments)])
```

**Reply by Munksgaard** [reply]
That's exactly what I needed, thank you very much!

**Reply by tarsius** [reply]
You're welcome.

This also got me to revisit an old kludge, see 732ec975ec0a9597c4bb4ce79c6040da72a66df2.
